examples/window-app/nrfconnect/main/WindowCovering.cpp

```diff
@@ -0,0 +1,297 @@
+#include <zephyr.h>
+
+LOG_MODULE_DECLARE(app, CONFIG_MATTER_LOG_LEVEL);
+
+using namespace ::chip::Credentials;
+using namespace ::chip::DeviceLayer;
+using namespace chip::app::Clusters::WindowCovering;
+
+static k_timer sMoveTimer;
+static constexpr uint32_t sMoveTimeoutMs{ 200 };
+
+static constexpr float FromOneRangeToAnother(uint32_t aInMin, uint32_t aInMax, uint32_t aOutMin, uint32_t aOutMax, uint32_t aInput)
+{
+    const auto diffInput  = aInMax - aInMin;
+    const auto diffOutput = aOutMax - aOutMin;
+    if (diffInput > 0)
+    {
+        return aOutMin + ((float) diffOutput / (float) diffInput) * (aInput - aInMin);
+    }
+    return 0.0f;
+}
+
+WindowCovering::WindowCovering() :
+    mLiftIndicator(LIFT_PWM_DEVICE, LIFT_PWM_CHANNEL, 0, 255), mTiltIndicator(TILT_PWM_DEVICE, TILT_PWM_CHANNEL, 0, 255)
+{
+    mLiftLED.Init(LIFT_STATE_LED);
+    mTiltLED.Init(TILT_STATE_LED);
+    mLiftIndicator.Init();
+    mTiltIndicator.Init();
+
+    k_timer_init(&sMoveTimer, MoveTimerTimeoutCallback, nullptr);
+}
+
+void WindowCovering::StartMove(MoveType aMoveType)
+{
+    mCurrentMoveType = aMoveType;
+    k_timer_start(&sMoveTimer, K_MSEC(sMoveTimeoutMs), K_NO_WAIT);
+}
+
+void WindowCovering::MoveTimerTimeoutCallback(k_timer * aTimer)
+{
+    if (!aTimer)
+        return;
+
+    chip::DeviceLayer::PlatformMgr().ScheduleWork(CallbackPositionSet);
+}
+
+void WindowCovering::CallbackPositionSet(intptr_t)
+{
+    EmberAfStatus status{};
+    chip::Percent100ths percent100ths{};
+    NPercent100ths current{};
+
+    if (Instance().mCurrentMoveType == MoveType::LIFT)
+    {
+        status = Attributes::CurrentPositionLiftPercent100ths::Get(Endpoint(), current);
+    }
+    else if (Instance().mCurrentMoveType == MoveType::TILT)
+    {
+        status = Attributes::CurrentPositionTiltPercent100ths::Get(Endpoint(), current);
+    }
+
+    if ((status == EMBER_ZCL_STATUS_SUCCESS) && !current.IsNull())
+    {
+        static constexpr auto sPercentDelta{ WC_PERCENT100THS_MAX_CLOSED / 50 };
+        percent100ths = ComputePercent100thsStep(OperationalStatusGet(Endpoint()).global, current.Value(), sPercentDelta);
+    }
+    else
+    {
+        LOG_ERR("Cannot read the current lift position. Error: %d", static_cast<uint8_t>(status));
+    }
+
+    NPercent100ths position{};
+    position.SetNonNull(percent100ths);
+    if (Instance().mCurrentMoveType == MoveType::LIFT)
+    {
+        LiftPositionSet(Endpoint(), position);
+    }
+    else if (Instance().mCurrentMoveType == MoveType::TILT)
+    {
+        TiltPositionSet(Endpoint(), position);
+    }
+
+    if (!TargetCompleted())
+    {
+        k_timer_start(&sMoveTimer, K_MSEC(sMoveTimeoutMs), K_NO_WAIT);
+    }
+}
+
+bool WindowCovering::TargetCompleted()
+{
+    NPercent100ths current{};
+    NPercent100ths target{};
+    EmberAfStatus status{};
+
+    if (Instance().mCurrentMoveType == MoveType::LIFT)
+    {
+        status = Attributes::CurrentPositionLiftPercent100ths::Get(Endpoint(), current);
```

Comment:
Why do we assign the status if we never check it? Either check or remove the assignment.